bionic/core/flow_execution.py

```diff
@@ -144,9 +136,16 @@ def _get_or_create_entry_for_state(self, state, is_requested=False):
             # Before doing anything with this task state, we should make sure its
             # cache state is up to date.
             state.refresh_all_persistent_cache_state(self._bootstrap)
+            dep_entries = [
```

Comment:
Not really. It's just easier to do it here in case we use this method someplace else in the future.

Reply:
So this doesn't change the number of entries that get created and refreshed? I would have guessed that this new code would recursively create and refresh every single ancestor of every entry we touch, whereas the old code only does this for entries we actually process, plus their immediate parents. But I haven't thought too deeply about this, maybe I'm wrong?

Reply:
If the question is around recursive vs using a stack, I can use a stack in this method instead. Wdyt?

Reply:
I'm not concerned about recursion in general. My concern is that this will create more entries and do more cache resets than the previous code.

(The only reason I care about using recursion for tasks is because we already need a stack-based approach for some cases, and I think having two approaches is complicated and gives up some of the benefits of the stack-based approach. But in general I think it's fine to recurse through the DAG.)

Reply:
Let's talk more about this in our daily check-in. I think we need to create all the entries if we are moving away from `TaskState` towards `TaskRunnerEntry`. But we can cut down on refreshing cache accessors by only calling them when used.

Reply:
Oh, I could buy that. Talking more sounds good!